Lib/_collections_abc.py

```diff
@@ -444,17 +444,15 @@ def __create_ga(cls, origin, args):
         return super().__new__(cls, origin, ga_args)
 
     def __repr__(self):
-        if len(self.__args__) == 2 and (self.__args__[0] is Ellipsis
-                                        or _concat_or_paramspec(self.__args__[0])):
+        if len(self.__args__) == 2 and _has_special_args(self.__args__[0]):
```

Comment:
If you want to you can move the `len() == 2` test into the helper too (just pass args instead of args[0]).